test/client_test.rb

```diff
@@ -19,10 +19,25 @@ def setup
     @host   = ThreeScale::Client::DEFAULT_HOST
   end
 
-  def test_raises_exception_if_provider_key_is_missing
+  def test_raises_exception_if_no_credentials_are_specified
     assert_raises ArgumentError do
       ThreeScale::Client.new({})
     end
+    assert_raises ArgumentError do
+      ThreeScale::Client.new(use_service_tokens: false)
+    end
+  end
+
+  def test_does_not_raise_if_some_credentials_are_specified
+    assert_nothing_raised ArgumentError do
+      ThreeScale::Client.new(provider_key: 'some_key')
+    end
+    assert_nothing_raised ArgumentError do
+      ThreeScale::Client.new(service_token: 'some_token')
+    end
+    assert_nothing_raised ArgumentError do
```

Comment:
assert_nothin_raised is noop and not necessary, especially giving it arguments just makes it look like there is some magic going on but it just does `yield`

better style is just asserting that the .new returns something or that you can get the value from attr reader